src/components/ContentOpenWith/__tests__/IntegrationPortal-test.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import IntegrationPortal from '../IntegrationPortal';
+
+const INTEGRATION_CONTAINER_ID = 'integration-container';
+
+describe('components/ContentOpenWith/IntegrationPortal', () => {
+    const getWrapper = props => shallow(<IntegrationPortal {...props} />);
+
+    it('should use an existing container if possible', () => {
+        const container = document.createElement('div');
+        container.id = INTEGRATION_CONTAINER_ID;
+        document.body.appendChild(container);
+
+        const wrapper = getWrapper({
+            integrationWindow: {
+                // use normal document in our integrationWindow
+                document,
+            },
+        });
+
+        const instance = wrapper.instance();
+        expect(instance.containerElement).toEqual(container);
+    });
+
+    it('should create the container if needed, and assign an id ', () => {
+        getWrapper({
+            integrationWindow: {
+                // use normal document in our integrationWindow
+                document,
+            },
+        });
+
+        const containerElement = document.querySelector('div');
+        expect(containerElement).toBeTruthy();
+        expect(containerElement.id).toEqual(INTEGRATION_CONTAINER_ID);
+    });
+
+    describe('componentDidMount()', () => {
+        it('should copy styles and append the container to the integration window', () => {
+            const wrapper = getWrapper({
```

Comment:
Can this be moved into a ```beforeEach```? Every test seems to be doing it

Reply:
the props do change a bit between cases.